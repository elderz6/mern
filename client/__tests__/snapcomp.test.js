import React from 'react'
import Enzyme,{ shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import AppNav from '../src/components/AppNav';
import toJSON from 'enzyme-to-json';

Enzyme.configure({adapter: new Adapter()});
function setup()
{
  const props =
  {
    state:{isOpen:false},
    toggle:jest.fn()
  };
  const enzymeWrapper = shallow(<AppNav {...props}/>);

  return{
    props,
    enzymeWrapper
  }
}

describe('Components with snapshot test', ()=>
{
  const {enzymeWrapper} = setup()
  it('should render correctly', ()=>
  {
    expect(toJSON(enzymeWrapper)).toMatchSnapshot();
    // console.log(toJSON(enzymeWrapper));
    
  });
});
