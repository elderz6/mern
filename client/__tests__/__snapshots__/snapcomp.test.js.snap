// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components with snapshot test should render correctly 1`] = `
<div
  className="jest"
>
  <Navbar
    className="mb-5"
    color="dark"
    dark={true}
    expand="sm"
    tag="nav"
  >
    <Container
      tag="div"
    >
      <NavbarBrand
        href="/"
        tag="a"
      >
        Shopping List
      </NavbarBrand>
      <NavbarToggler
        onClick={[Function]}
        tag="button"
        type="button"
      />
      <Collapse
        appear={false}
        enter={true}
        exit={true}
        in={false}
        isOpen={false}
        mountOnEnter={false}
        navbar={true}
        onEnter={[Function]}
        onEntered={[Function]}
        onEntering={[Function]}
        onExit={[Function]}
        onExited={[Function]}
        onExiting={[Function]}
        tag="div"
        timeout={350}
        unmountOnExit={false}
      >
        <Nav
          className="ml-auto"
          navbar={true}
          tag="ul"
          vertical={false}
        >
          <NavItem
            tag="li"
          >
            <LogForm />
          </NavItem>
          <NavItem
            tag="li"
          >
            <NavLink
              href="https://github.com/elderz6"
              tag="a"
            >
              GitHub
            </NavLink>
          </NavItem>
        </Nav>
      </Collapse>
    </Container>
  </Navbar>
</div>
`;
